pkg/controller/util/controllerconfig.go

```diff
@@ -44,7 +46,7 @@ type LeaderElectionConfiguration struct {
 	RetryPeriod time.Duration
 	// resourceLock indicates the resource object type that will be used to lock
 	// during leader election cycles.
-	ResourceLock string
+	ResourceLock fedv1a1.ResourceLockType
```

Comment:
(No action required) I think this struct predates `FederationConfig`.  I think the use of `ControllerConfig` should either be replaced by `FederationConfig` or at least reuse structs like `LeaderElectConfig` to avoid having to manually synchronize them.